### PR TITLE
Helm support

### PR DESCRIPTION
**Pull request by Pluies**
#### What type of PR is this?

/kind documentation
/kind feature

#### What this PR does / why we need it:

Add support for Helm as an upstream! 🎉

Also big README updates, and fixes a tiny bug on the gitlab provider
where repositories containing a slash were broken.

Also added a `test-docker` command to the makefile, to allow testing in
a "neutral" environment without being impacted by local git settings.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
New upstream: helm
```

### CODE REVIEW

**Comment by Pluies** [comment]
Redo of #44 , but with tests this time, and from my fork rather than the repo itself 👍

File: upstream/helm.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+// for the given repository (depending on the Constraints if set).
+func (upstream Helm) LatestVersion() (string, error) {
+	log.Debug("Using Helm flavour")
+	return latestChartVersion(upstream)
+}
+
+func latestChartVersion(upstream Helm) (string, error) {
+	// Sanity checking
+	if upstream.Repo == "" {
+		return "", errors.Errorf("Invalid helm upstream: missing repo argument")
```
Comment (saschagrunert) [inline upstream/helm.go:58]: nit: let's always start errors lower-case 
```suggestion
		return "", errors.Errorf("invalid helm upstream: missing repo argument")
```

File: upstream/helm.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	}
+
+	if upstream.Chart == "" {
+		return "", errors.Errorf("Invalid helm upstream: missing chart argument")
+	}
+
+	if !strings.Contains(upstream.Repo, "//") {
```
Comment (saschagrunert) [inline upstream/helm.go:65]: Do we wanna be more constraint here with:
```suggestion
	if !strings.HasPrefix(upstream.Repo, "https://") || !strings.HasPrefix(upstream.Repo, "http://") {
```

**Review by saschagrunert — APPROVED** [review]
Just two nits, otherwise LGTM

**Review by cpanato — APPROVED** [review]
/lgtm thanks for this
just the comments from Sascha

**Comment by Pluies** [comment]
Thank you both for the reviews!

@saschagrunert I've fixed the error messages, and tightened the schema detection as per your suggestion. Helm supports [http, https and oci by default](https://github.com/helm/helm/blob/29d273f985306bc508b32455d77894f3b1eb8d4d/pkg/getter/getter.go#L156-L174), so I'm now checking both that we have a valid URL, and the scheme is one of these 👍

**Review by saschagrunert — APPROVED** [review]
/unhold

**Pull request merged**